### PR TITLE
Remove old code examples.

### LINKED ISSUE #114: autodoc references are broken because packages aren't installed
### Issue Description

The following warning appears in ReadTheDocs build logs:
```
WARNING: autodoc: failed to import function 'get_database' from module 'signac'; the following exception was raised:
No module named 'signac'
WARNING: autodoc: failed to import method 'Project.create_linked_view' from module 'signac'; the following exception was raised:
No module named 'signac'
```

### Proposed change (optional)

This is happening because signac isn't installed by the signac-docs ReadTheDocs config. However, we shouldn't be using autodoc directly in the signac docs at all, in my opinion. We should only use links/references to the signac API docs instead. I propose deleting these (we already need to overhaul the indexing page, since many of its features are deprecated).

### PR DESCRIPTION
**Pull request by bdice**
## Description
I removed old code examples. These are no longer referenced in the docs and are no longer showing good practices.

Resolves #114.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/master/ContributorAgreement.md).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

**Comment by bdice** [comment]
CircleCI will fail on this PR -- that will be fixed in #117.

File: docs/source/indexing.rst [hunk 1]
```diff
@@ -304,9 +304,6 @@ Database access
 
 After :doc:`configuring <configuration>` one or more database hosts you can access a database with the :py:func:`signac.get_database` function.
```
Comment (atravitz) [inline docs/source/indexing.rst:305]: Is this deprecated? If not, we should improve these docs in a separate PR

**Review by atravitz — APPROVED** [review]
looks good to me!

File: docs/source/indexing.rst (hunk 1 above)
Comment (bdice) [reply docs/source/indexing.rst:305]: We'll need to do that sometime around the release of signac 2.0. This entire page will probably be removed.

**Pull request merged**